study1/week4/ValidPalindrome2.java

```diff
@@ -0,0 +1,38 @@
+class Solution {
+    public boolean validPalindrome(String s) {
+        int end = s.length() - 1;
+
+        for(int i = 0; i < s.length()/2 ; i++){
```

Comment:
you have two control variables in a loop, it might be better to understand the code by making this a while loop, or you can do the following. Also, `i< s.length() / 2`  is same as `begin < end` .
```java
for (int start = 0, end = 0; start < end ; ++start, --end) {
....
}
```
or with while loop
```java
while (start < end) {
...
++start;
--end;
}
```